src/com/goide/inspections/GoExportedOwnDeclarationInspection.java

```diff
@@ -0,0 +1,207 @@
+    holder.registerProblem(namedElement, errorText, ProblemHighlightType.WEAK_WARNING, new MyLocalQuickFixBase(index));
+  }
+
+  private static class MyLocalQuickFixBase extends LocalQuickFixBase {
+    private int index;
+
+    public MyLocalQuickFixBase(int index) {
+      super(QUICK_FIX_NAME);
+      this.index = index;
+    }
+
+    @Override
+    public void applyFix(@NotNull final Project project, @NotNull ProblemDescriptor descriptor) {
+      final PsiElement element = descriptor.getPsiElement();
+      if (element == null) return;
+      new WriteCommandAction.Simple(project, getName(), element.getContainingFile()) {
+        @Override
+        protected void run() throws Throwable {
+          List<GoExpression> elementValueList;
+          String elementType = "";
+          if (element instanceof GoConstDefinition) {
+            elementValueList = ((GoConstSpec)element.getParent()).getExpressionList();
+            GoType elemType = ((GoConstSpec)element.getParent()).getType();
+            if (elemType != null) {
+              elementType = elemType.getText();
+            }
+          }
+          else if (element instanceof GoVarDefinition) {
+            elementValueList = ((GoVarSpec)element.getParent()).getExpressionList();
+            GoType elemType = ((GoVarSpec)element.getParent()).getType();
+            if (elemType != null) {
+              elementType = elemType.getText();
+            }
+          }
+          else {
+            return;
+          }
+
+          if (elementValueList.size() != 0 &&
+              elementValueList.size() <= index) {
+            return;
+          }
+
+          PsiElement elementParent = element.getParent();
+          PsiElement newElement;
+          PsiElement afterElement;
+          if (elementValueList.size() > 0) {
+            PsiElement elementValue = elementValueList.get(index);
+            if (element instanceof GoConstDefinition) {
+              newElement = GoElementFactory.createConstSpec(project, element.getText(), elementType, elementValue.getText());
+            }
+            else {
+              newElement = GoElementFactory.createVarSpec(project, element.getText(), elementType, elementValue.getText());
+            }
+          }
+          else {
+            if (element instanceof GoConstDefinition) {
+              newElement = GoElementFactory.createConstSpec(project, element.getText(), elementType, "");
+            }
+            else {
+              newElement = GoElementFactory.createVarSpec(project, element.getText(), elementType, "");
+            }
+          }
+
+          afterElement = getAfterElement(element);
+          PsiElement grandParent = elementParent.getParent();
+          elementParent.addAfter(newElement, afterElement);
+          elementParent.addAfter(GoElementFactory.createNewLine(project), afterElement);
+
+          if (elementParent instanceof GoConstSpec) {
```

Comment:
So you didn't know what elementParent until now, but you've already added some elements there, this is looks very brittle